### PR TITLE
[Refactor] Merge With Shisen OpenCV

### PR DESCRIPTION
**Pull request by Nathan24zz**
## Jira Link: 
https://ichiro-its.atlassian.net/browse/PI-181

## Description
Merge shisen_cpp with shisen_opencv and refactor the package to a new system design. Now if other packages want to get an image from the webcam, it doesn't have to subscribe to ros topic but just include ImageProvider.

## Type of Change

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause the existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:

- [ ] Using Branch Name Convention
    - `feature/JIRA-ID-SHORT-DESCRIPTION` if has a JIRA ticket
    - `enhancement/SHORT-DESCRIPTION` if has/has no JIRA ticket and contain enhancement
    - `hotfix/SHORT-DESCRIPTION` if the change doesn't need to be tested (urgent)
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made the documentation for the corresponding changes.

### CODE REVIEW

File: .github/workflows/build-debian-stable.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ jobs:
       - name: Add stable Debian repository and rosdep sources list
```
Comment (maroqijalil) [inline .github/workflows/build-debian-stable.yml:15]: Check my comment on `ninshiki_interfaces` regarding its wf.

File: include/shisen_cpp/viewer/consumer/image_consumer.hpp [hunk 2]
```diff
@@ -18,43 +18,36 @@
...
 #include <string>
 
-#include "../node.hpp"
+#include "shisen_cpp/utility.hpp"
 
-namespace shisen_cpp
+
+namespace shisen_cpp::viewer
 {
+using Image = shisen_interfaces::msg::Image;
```
Comment (maroqijalil) [inline include/shisen_cpp/viewer/consumer/image_consumer.hpp]: Declare inside the class instead. Also, check the other occurrences.

File: include/shisen_cpp/viewer/consumer/image_consumer.hpp [hunk 3]
```diff
@@ -18,43 +18,36 @@
...
-  struct Options : public virtual CameraNode::Options
-  {
-  };
-
-  explicit ImageConsumer(
-    rclcpp::Node::SharedPtr node, const Options & options = Options());
-
-  ~ImageConsumer();
-
   virtual void on_image_changed(const Image & image);
```
Comment (maroqijalil) [inline include/shisen_cpp/viewer/consumer/image_consumer.hpp]: Why did you add `virtual` specifier here? Please also check the other declared method with virtual specifier.

File: include/shisen_cpp/viewer/consumer/image_consumer.hpp (hunk 2 above)
Comment (maroqijalil) [reply include/shisen_cpp/viewer/consumer/image_consumer.hpp:36]: Declare inside the class instead. Please also check the other declared method with `virtual` specifier.

File: .github/workflows/build-debian-stable.yml (hunk 1 above)
Comment (Nathan24zz) [reply .github/workflows/build-debian-stable.yml:15]: done

File: include/shisen_cpp/viewer/consumer/image_consumer.hpp (hunk 2 above)
Comment (Nathan24zz) [reply include/shisen_cpp/viewer/consumer/image_consumer.hpp]: done

File: include/shisen_cpp/viewer/consumer/image_consumer.hpp (hunk 3 above)
Comment (Nathan24zz) [reply include/shisen_cpp/viewer/consumer/image_consumer.hpp]: I forgot to delete that after refactoring from the previous
